schedule-replicas-by-topology-labels.md

```diff
@@ -0,0 +1,131 @@
+tikv-3 labels="zone=z1,host=h2"
+tikv-4 labels="zone=z1,host=h2"
+# z2
+tikv-5 labels="zone=z2,host=h1"
+tikv-6 labels="zone=z2,host=h2"
+# z3
+tikv-7 labels="zone=z3,host=h1"
+tikv-8 labels="zone=z3,host=h2"
+
+[pd_servers:vars]
+location_labels = ["zone", "host"]
+```
+
+## PD schedules based on topology label
+
+PD schedules replicas according to the label level to make sure that different replicas of the same data are distributed as much as possible.
+
+Take the topology in the previous section as an example.
+
+Assume that the number of cluster replicas is 3 (`max-replicas=3`). As there are 3 zones in total, PD keeps the 3 replicas of each Region placed in z1/z2/z3, so that the TiDB cluster is still available when any data center fails,
+
+Assume that the number of cluster replicas is 5 (`max-replicas=5`). As there are only 3 zones in total, PD can not guarantee the isolation of each replica at this level. At this time, the PD scheduler adjusts to ensure isolation at the host level. In other words, multiple copies of a Region might be distributed in the same zone, but will not be distributed on the same host.
+
+Under the premise of the 5-replica configuration, if z3 occurs an overall failure or isolation, and can not be restored in a period (controlled by `max-store-down-time`), PD will make up the 5 replicas through scheduling. At this time, only 3 hosts are available, so host level isolation can not be guaranteed. Therefore, multiple copies might be distributed to the same host. But if the value `isolation-level` is set to `zone` instead of empty, this specifies the minimum physical isolation requirements for Region replicas. That is to say, PD ensures that replicas of the same Region are scattered in different zones. Even if following this isolation restriction does not meet the multiple replica requirements of `max-replicas`, PD will not perform corresponding scheduling. 
+
+For example, there are three data centers z1, z2 and z3 in the TiKV cluster. Under the three-replica setting, PD distributes the three replicas of the same Region to these three data centers respectively. At this time, if the entire data center z1 encounters a power outage and cannot be recovered in a period, the PD will consider that the Region copy on z1 is no longer available. But because the `isolation-level` is set to `zone`, PD needs to strictly ensure that different Region replicas will not fall into the same zone. At this time, as both z2 and z3 already have replicas, PD does not perform any scheduling under the restriction of the minimum mandatory isolation level, even if there are only two replicas.
+
+Similarly, when `isolation-level` is set to `rack`, the minimum isolation level is different racks in the same data center. Under this setting, if isolation can be guaranteed at the zone level, it will be guaranteed first. Only when the zone level isolation can not be completed, will the scheduling in the same zone and the same rack be considered to avoid.
```

Comment:
```suggestion
Similarly, when `isolation-level` is set to `rack`, the minimum isolation level applies to different racks in the same data zone. With this configuration, the isolation at the zone level is guaranteed first if possible. When the isolation at the zone level cannot be guaranteed, PD tries to avoid scheduling different replicas to the same rack in the same zone, and so on.
```

Reply:
It's not recommended to use "and so on" or "etc" according to[ the google style guide](https://developers.google.com/style/word-list#letter-e) @TomShawn 
> Avoid both etc. and and so on wherever possible, but if you really need to use one, use etc. Always include the period, even if a comma follows immediately after.

Reply:
Which one shall we choose? "data zone" or "data center"
see @disksing 's [comment](https://github.com/pingcap/docs/pull/3825#discussion_r484268468)

Reply:
@ireneontheway Comment changed, PTAL.

Reply:
I think data center is more widely used.

Reply:
```suggestion
Similarly, when `isolation-level` is set to `rack`, the minimum isolation level is different racks in the same data center. Under this setting, if isolation can be guaranteed at the zone layer, it will be guaranteed first. Only when the zone level isolation can not be completed, will the scheduling in the same zone and the same rack be considered to avoid.
```
Is it ok to change "level" into "layer"?